orthopairs/src/main/java/org/reactome/release/orthopairs/Main.java

```diff
@@ -0,0 +1,155 @@
+            } else {
+                logger.info(altIdFilepath + " already exists");
+            }
+        }
+
+        JSONParser parser = new JSONParser();
+        JSONObject speciesJSONFile = (JSONObject) parser.parse(new FileReader(pathToSpeciesConfig));
+
+        // This method will produce two multi-level Maps, sourceTargetProteinHomologs and targetGeneProteinMap
+        // sourceTargetProteinHomologs structure: {TargetSpecies-->{SourceProteinId-->[TargetHomologousProteinIds]}}
+        // targetGeneProteinMap structure: {TargetSpecies-->{TargetGeneId-->[targetProteinIds]}}
+        // The lower-level structure is a Set to reduce redundancy.
+        OrthologyFileParser.parsePantherOrthologFiles(pantherFiles, sourceMappingSpecies, speciesJSONFile);
+        Map<String,Map<String,Set<String>>> sourceTargetProteinHomologs = OrthologyFileParser.getSourceAndTargetProteinHomologs();
+        Map<String,Map<String,Set<String>>> targetGeneProteinMap = OrthologyFileParser.getTargetGeneProteinMap();
+        // Produces the protein homology and species gene-protein files
+        for (Object speciesKey : speciesJSONFile.keySet()) {
+            // No point in the source species mapping to itself
+            if (!speciesKey.equals(sourceMappingSpecies)) {
+                JSONObject speciesJSON = (JSONObject)speciesJSONFile.get(speciesKey);
+                JSONArray speciesNames = (JSONArray) speciesJSON.get("name");
+                logger.info("Attempting to create orthopairs files for " + speciesNames.get(0));
+                String speciesPantherName = speciesJSON.get("panther_name").toString();
+                // Produces the {sourceSpecies}_{targetspecies}_mapping.txt file
+                String sourceTargetProteinMappingFilename = releaseNumber + "/" + sourceMappingSpecies + "_" + speciesKey + "_mapping.txt";
+                Map<String,Set<String>> speciesProteinHomologs = sourceTargetProteinHomologs.get(speciesPantherName);
+                OrthopairFileGenerator.createProteinHomologyFile(speciesKey.toString(), sourceTargetProteinMappingFilename, speciesJSON, speciesProteinHomologs);
+                // Produces the {targetSpecies}_gene_protein_mapping.txt file
+                String targetGeneProteinMappingFilename = releaseNumber + "/" + speciesKey + "_gene_protein_mapping.txt";
+                Map<String,Set<String>> speciesGeneProteinMap = targetGeneProteinMap.get(speciesPantherName);
+                OrthopairFileGenerator.createSpeciesGeneProteinFile(speciesKey.toString(), targetGeneProteinMappingFilename, speciesJSON, speciesGeneProteinMap);
+            }
+        }
+        logger.info("Finished Orthopairs file generation");
+    }
+
+    private static void downloadAndExtractTarFile(String pantherFilename, String pantherFilepath) throws IOException {
+
+        URL pantherFileURL = new URL(pantherFilepath + pantherFilename);
+        File pantherTarFile = new File(pantherFilename);
+
+        // Download files
```

Comment:
This is the other spot where a download method could be called if extracted out.